src/basic/fs-util.c

```diff
@@ -195,17 +195,17 @@ int readlink_and_make_absolute(const char *p, char **r) {
         return 0;
 }
 
-int chmod_and_chown(const char *path, mode_t mode, uid_t uid, gid_t gid) {
+int chmod_and_chown_at(int dir_fd, const char *path, mode_t mode, uid_t uid, gid_t gid) {
         _cleanup_close_ int fd = -1;
 
```

Comment:
assert on dird_fd

Reply:
`openat()` like calls generally don't put any requirements on `dir_fd`, it can be negative, `AT_FDCWD` or positive and all of those mean different things, so I don't think we should assert here.